lib/puppet/util/windows/win32/process.rb

```diff
@@ -0,0 +1,428 @@
+    # process_handle and the thread_handle are automatically closed for you
+    # before the ProcessInfo struct is returned.
+    #
+    # If the 'with_logon' option is set, then the process runs the specified
+    # executable file in the security context of the specified credentials.
+
+    VALID_KEYS = %i[
+      app_name command_line inherit creation_flags cwd environment
+      startup_info thread_inherit process_inherit close_handles with_logon
+      domain password
+    ].freeze
+
+    VALID_SI_KEYS = %i[
+      startf_flags desktop title x y x_size y_size x_count_chars
+      y_count_chars fill_attribute sw_flags stdin stdout stderr
+    ].freeze
+
+    private_constant :VALID_KEYS, :VALID_SI_KEYS
+
+    def create(args)
+      # Validate that args is a Hash
+      validate_args(args)
+
+      initialize_defaults
+
+      # Validate the keys, and convert symbols and case to lowercase strings.
+      validate_keys(args)
+
+      # If the startup_info key is present, validate its subkeys
+      validate_startup_info if hash[:startup_info]
+
+      # validates that 'app_name' or 'command_line' is set
+      validate_command_line
+
+      if hash[:app_name] && !hash[:command_line]
+        hash[:command_line] = hash[:app_name]
+        hash[:app_name] = nil
+      end
+
+      # Setup stdin, stdout and stderr handlers
+      setup_std_handlers
+
+      if logon
+        create_process_with_logon
+      else
+        create_process
+      end
+
+      # Automatically close the process and thread handles in the
+      # PROCESS_INFORMATION struct unless explicitly told not to.
+      if hash[:close_handles]
+        FFI::WIN32.CloseHandle(procinfo[:hProcess])
+        FFI::WIN32.CloseHandle(procinfo[:hThread])
+      end
+
+      ProcessInfo.new(
+        procinfo[:hProcess],
+        procinfo[:hThread],
+        procinfo[:dwProcessId],
+        procinfo[:dwThreadId]
+      )
+    end
+
+    remove_method :setpriority
+
+    # Sets the priority class for the specified process id +int+.
+    #
+    # The +kind+ parameter is ignored but present for API compatibility.
+    # You can only retrieve process information, not process group or user
+    # information, so it is effectively always Process::PRIO_PROCESS.
+    #
+    # Possible +int_priority+ values are:
+    #
+    # * Process::NORMAL_PRIORITY_CLASS
+    # * Process::IDLE_PRIORITY_CLASS
+    # * Process::HIGH_PRIORITY_CLASS
+    # * Process::REALTIME_PRIORITY_CLASS
+    # * Process::BELOW_NORMAL_PRIORITY_CLASS
+    # * Process::ABOVE_NORMAL_PRIORITY_CLASS
+
+    def setpriority(kind, int, int_priority)
+      raise TypeError unless kind.is_a?(Integer)
+      raise TypeError unless int.is_a?(Integer)
+      raise TypeError unless int_priority.is_a?(Integer)
+
+      int = Process.pid if int == 0
+      handle = FFI::WIN32::OpenProcess(PROCESS_SET_INFORMATION, 0 , int)
+
+      if handle == 0
+        raise SystemCallError, FFI.errno, "OpenProcess"
+      end
```

Comment:
We've been using `Puppet::Util::Windows::Error` as it generates a human readable error message, but since we're just removing win32-process usage, this is fine for now.

Reply:
Using `Puppet::Util::Windows::Error` will force us to load all `util/windows`